src/Symfony/Component/Polyfill/Functions/Php70.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Polyfill\Functions;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+class Php70
+{
+    public static function random_bytes($length)
+    {
+        if (0 >= $length) {
+            // PHP 7.0 throws an \Error. E_USER_ERROR is the best we can do here.
+            trigger_error('Length must be greater than 0', E_USER_ERROR);
+
+            // This should never be reached: E_USER_ERROR should break code execution.
+            throw new \InvalidArgumentException('Length must be greater than 0');
+        }
+
+        $length = (int) $length;
+        $allBytes = '';
+
+        if (function_exists('mcrypt_create_iv') && $bytes = @mcrypt_create_iv($length, MCRYPT_DEV_URANDOM)) {
+            if (isset($bytes[$length - 1])) {
+                return $bytes;
+            }
+            $allBytes .= $bytes;
+        }
+        if (function_exists('Sodium\randombytes_buf') && $bytes = \Sodium\randombytes_buf($length)) {
+            if (isset($bytes[$length - 1])) {
+                return $bytes;
+            }
+            $allBytes .= $bytes;
+        }
+        if ('\\' !== DIRECTORY_SEPARATOR && $h = @fopen('/dev/urandom', 'rb')) {
+            if (function_exists('stream_set_read_buffer')) {
+                stream_set_read_buffer($h, 8);
+            }
+            $bytes = fread($h, $length);
+            $stat = fstat($h);
+            fclose($h);
+            if (isset($bytes[$length - 1]) && ($stat['mode'] & 0170000) === 020000) {
```

Comment:
:+1: for adopting the same sanity checks that @jedisct1 and @narfbg provided random_compat.